File: src/main/java/Lec7_Exception.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import org.jetbrains.annotations.NotNull;

/**
 * Created by jihun.park@navercorp.com on 2022-09-27.
 */
public class Lec7_Exception {

    public static void main(String[] args) throws IOException {
        readFile("/");
    }

    private int parseIntOrThrow(@NotNull String str){
        try{
            return Integer.parseInt(str);
        }catch(NumberFormatException e){
            throw new IllegalArgumentException(String.format("주어진 %s는 숫자가 아닙니다.",str));
        }
    }
    private Integer parseIntOrThrowV2(@NotNull String str){
        try{
            return Integer.parseInt(str);
        }catch(NumberFormatException e){
            return null;
        }
    }

    public static void readFile() throws IOException {
        File currentFile = new File(".");
        File file = new File(currentFile.getAbsolutePath() + "/a.txt");
        BufferedReader reader = new BufferedReader(new FileReader(file));
        System.out.println(reader.readLine());
        reader.close();

    }

    public static void readFile(String path) throws IOException {
        try (BufferedReader reader = new BufferedReader(new FileReader(path))) {
            System.out.println(reader.readLine());
    }
}

}

File: src/main/kotlin/Lec7_Exception.kt
import java.io.BufferedReader
import java.io.File
import java.io.FileReader

/**
 * Created by jihun.park@navercorp.com on 2022-09-27.
 */

fun parseIntOrThrow(str :String) : Int{
    try{
        return str.toInt()
    }catch(e: NumberFormatException){
        throw IllegalArgumentException("주어진 ${str}는 숫자가 아닙니다.")
    }
}

fun parseIntOrThrow2(str :String) : Int?{   // 하나의 표현식 처럼 처리
    return try{
        str.toInt()
    }catch(e: NumberFormatException){
        null
    }
}

fun readFile(){
    val currentFile = File(".")
    val file = File(currentFile.absolutePath + "/a.txt")
    val reader = BufferedReader(FileReader(file))
    println(reader.readLine())
    reader.close()
}

fun readFile(path: String){
    BufferedReader(FileReader(path)).use { reader ->
        println(reader.readLine())
    }
}